### PR TITLE
StreakService for streak computation

### LINKED ISSUE #385: Streak Computation Service
Create `StreakService.kt` that handles streak logic:

- Determine "week" boundaries (Monday 00:00 UTC to Sunday 23:59 UTC)
- When a user joins an Event or Serie in a group, call `onActivityJoined(groupId, userId, activityDate)`
- Increment `currentStreakActivities`; if new week, increment `currentStreakWeeks`
- If `lastActiveWeekStart` is >1 week ago, reset current streak before incrementing
- Update `bestStreak*` fields if current exceeds best

### PR DESCRIPTION
**Pull request by BryanDMP**
# Add StreakService for streak computation and lifecycle management

## Overview

This PR implements the complete streak computation system for tracking individual user activity streaks within groups. Streaks are maintained by participating in group activities across consecutive weeks.

### Core Streak Logic (`StreakService.kt`)
- **Week boundaries**: Monday 00:00 UTC to Sunday 23:59 UTC
- **`onActivityJoined(groupId, userId, activityDate)`**: Called when a user joins/creates a group event or serie
  - Increments `currentStreakActivities`
  - Increments `currentStreakWeeks` if activity is in a new consecutive week
  - Resets current streak if `lastActiveWeekStart` is >1 week ago
  - Updates `bestStreak*` fields when current exceeds best
- **`onActivityLeft(groupId, userId, activityDate)`**: Called when a user quits an event or serie
- **`onActivityDeleted(groupId, userIds, activityDate)`**: Called when an activity is deleted (reverts streaks for all participants)
- **`onUserLeftGroup(groupId, userId)`**: Deletes user's streak when leaving a group
- **`onGroupDeleted(groupId)`**: Deletes all streaks when a group is deleted

### Repository Layer
- Added `deleteStreakForUser(groupId, userId)` to clean up streak data when user leaves group
- Added `deleteAllStreaksForGroup(groupId)` to clean up all streaks when group is deleted

### ViewModel Integrations
| ViewModel | Method | Streak Call |
|-----------|--------|-------------|
| `CreateEventViewModel` | `createEvent()` | `onActivityJoined` for all group members |
| `CreateSerieViewModel` | `createSerie()` | `onActivityJoined` for all group members |
| `ShowEventViewModel` | `toggleParticipation()` | `onActivityJoined` (join) / `onActivityLeft` (quit) |
| `SerieDetailsViewModel` | `joinSerie()` / `quitSerie()` | `onActivityJoined` / `onActivityLeft` |
| `ShowEventViewModel` | `deleteEvent()` | `onActivityDeleted` for upcoming group events |
| `SerieDetailsViewModel` | `deleteSerie()` | `onActivityDeleted` for upcoming group series |
| `GroupListViewModel` | `leaveGroup()` | `onUserLeftGroup` |
| `GroupListViewModel` | `deleteGroup()` | `onGroupDeleted` |

## Testing

- **StreakServiceTest.kt**: 22 tests covering week boundary calculations, all streak operations, and edge cases
- **GroupStreakRepositoryFirestoreTest.kt**: Updated with tests for new delete methods
- **GroupStreakRepositoryLocalTest.kt**: Updated with tests for new delete methods
- **ShowEventViewModelTest.kt**: Added streak tests for `toggleParticipation()` and `deleteEvent()`.
- **SerieDetailsViewModelTest.kt**: Added streak tests for `joinSerie()` / `quitSerie()` and `deleteSerie()`.

Tests were optimized to maximize line coverage and minimize CI execution time. 

## PR Size Justification

This PR appears large in terms of diff size, but the scope is tightly controlled and the changes are structurally necessary. The streak functionality touches the three core domains of the app; groups, events, and series which means that adding backend streak computation required integrating logic into multiple view models. In several view models , new if / try / catch blocks pushed methods above the allowed cognitive-complexity threshold in SonarCloud, so **I performed targeted refactors to split heavy methods into focused helpers.**

In parallel, this PR includes approximately 1100 lines of tests, which naturally increases the overall diff size but keeps production code safe and maintainable. The backend work was deliberately split across two PRs:
	1.	the StreakRepository interfaces + implementations (Firestore + Local) and their tests
	2.	this PR, focused on streak computation, lifecycle management, and full integration

Given these constraints, this PR represents the smallest coherent unit of work that maintains correctness, respects SonarCloud, and keeps the architecture clean.

## What's Next

- View model + UI components to display streak information (leaderboard, user profile streak badge)

## AI Acknowledgment

Parts of `StreakService.kt` and the test files were developed with assistance from AI tools.

---

Closes #385

### CODE REVIEW

File: app/src/main/java/com/android/joinme/ui/groups/GroupListViewModel.kt [hunk 1]
```diff
@@ -94,6 +98,14 @@ class GroupListViewModel(
...
 
+        // Delete all streaks for this group before deleting the group
+        try {
+          StreakService.onGroupDeleted(groupId)
+        } catch (e: Exception) {
+          Log.e("GroupListViewModel", "Error deleting streaks for group $groupId", e)
+          // Non-critical: continue with group deletion even if streak deletion fails
+        }
+
         groupRepository.deleteGroup(groupId, currentUserId)
```
Comment (AlexisPDS) [inline app/src/main/java/com/android/joinme/ui/groups/GroupListViewModel.kt]: I think you should first delete the Group and then delete the Streak related to the Group, because if Streak deletion suceed and group deletion fails then we have a data loss.

File: app/src/main/java/com/android/joinme/ui/overview/CreateEventViewModel.kt [hunk 2]
```diff
@@ -118,126 +130,245 @@ class CreateEventViewModel(
...
+   */
+  private suspend fun updateStreaksSafe(group: Group, date: Timestamp) {
+    for (memberId in group.memberIds) {
       try {
-        repository.deleteEvent(event.eventId)
-      } catch (deleteError: Exception) {
-        Log.e("CreateEventViewModel", "Failed to rollback event creation", deleteError)
+        StreakService.onActivityJoined(group.id, memberId, date)
+      } catch (e: Exception) {
+        Log.e("CreateEventViewModel", "Error updating streak for user $memberId", e)
```
Comment (AlexisPDS) [inline app/src/main/java/com/android/joinme/ui/overview/CreateEventViewModel.kt]: If the Streak aren't updated Correcty you should thow an error to make it consistent. In the current implementation if it fails the user will not have his Streak updated and don't know why.

File: app/src/main/java/com/android/joinme/ui/overview/ShowEventViewModel.kt [hunk 3]
```diff
@@ -259,4 +203,187 @@ class ShowEventViewModel(
...
+  private suspend fun updateStreakSafe(event: Event, userId: String, isJoining: Boolean) {
+    if (event.groupId != null) {
+      try {
+        if (isJoining) {
+          StreakService.onActivityJoined(event.groupId, userId, event.date)
+        } else {
+          StreakService.onActivityLeft(event.groupId, userId, event.date)
+        }
+      } catch (e: Exception) {
+        Log.e("ShowEventViewModel", "Error updating streak for user $userId", e)
```
Comment (AlexisPDS) [inline app/src/main/java/com/android/joinme/ui/overview/ShowEventViewModel.kt]: If the Streak aren't updated Correcty you should thow an error to make it consistent. In the current implementation if it fails the user will not have his Streak updated and don't know why.

File: app/src/main/java/com/android/joinme/ui/overview/SerieDetailsViewModel.kt [hunk 4]
```diff
@@ -212,6 +217,16 @@ class SerieDetailsViewModel(
         // Update in repository
         seriesRepository.editSerie(serie.serieId, updatedSerie)
 
+        // Update streak for group series
+        if (serie.groupId != null) {
+          try {
+            StreakService.onActivityJoined(serie.groupId, currentUserId, serie.date)
+          } catch (e: Exception) {
+            Log.e("SerieDetailsViewModel", "Error updating streak for user $currentUserId", e)
+            // Non-critical: don't fail join operation if streak update fails
```
Comment (AlexisPDS) [inline app/src/main/java/com/android/joinme/ui/overview/SerieDetailsViewModel.kt]: If the Streak aren't updated Correcty you should thow an error to make it consistent. In the current implementation if it fails the user will not have his Streak updated and don't know why.

File: app/src/main/java/com/android/joinme/ui/overview/SerieDetailsViewModel.kt [hunk 5]
```diff
@@ -263,6 +280,16 @@ class SerieDetailsViewModel(
       // Update in repository
       seriesRepository.editSerie(serie.serieId, updatedSerie)
 
+      // Update streak for group series
+      if (serie.groupId != null) {
+        try {
+          StreakService.onActivityLeft(serie.groupId, currentUserId, serie.date)
+        } catch (e: Exception) {
+          Log.e("SerieDetailsViewModel", "Error updating streak for user $currentUserId", e)
```
Comment (AlexisPDS) [inline app/src/main/java/com/android/joinme/ui/overview/SerieDetailsViewModel.kt]: If the Streak aren't updated Correcty you should thow an error to make it consistent. In the current implementation if it fails the user will not have his Streak updated and don't know why.

**Review by AlexisPDS — APPROVED** [review]
Thanks for all these changes.
Looks good to me. 
Ready to merge.
